tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -352,6 +355,90 @@ def test_5_inc_snapshots(network_config,
+DEFAULT_TAP_NAME = "tap0"
+DEFAULT_DEV_NAME = "eth0"
+DEFAULT_NETMASK = 30
+
+
+def _get_process_start_time(p_pid):
+    # Helper function returning:
+    # - start time of process with provided PID
+    # - return code of stat -c%X /proc/$$ command
+    proc = subprocess.Popen(["stat -c%X /proc/{}".format(p_pid)],
+                            stdout=subprocess.PIPE,
+                            shell=True,
+                            encoding='ascii')
+    out = proc.communicate()[0].strip('\n')
+    return out, proc.returncode
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_load_snapshot_failure_handling(bin_cloner_path):
+    """
+    Test scenario.
+
+    1. Create two empty files representing snapshot memory and
+    microvm state
+    2. Try to build a VM out of the empty files.
+    3. Verify that an error was shown and the FC process is dead.
+    """
+    logger = logging.getLogger("snapshot_load_failure")
+
+    # Create a new VM instance
+    vm = MicrovmBuilder(bin_cloner_path).create_basevm()
+    vm.spawn(log_level='Info')
+
+    # Create two empty files for snapshot state and snapshot memory
+    chroot_path = vm.jailer.chroot_path()
+    snapshot_dir = os.path.join(chroot_path, "snapshot")
+    Path(snapshot_dir).mkdir(parents=True, exist_ok=True)
+
+    snapshot_mem = os.path.join(snapshot_dir, "snapshot_mem")
+    open(snapshot_mem, "w+").close()
+    snapshot_vmstate = os.path.join(snapshot_dir, "snapshot_vmstate")
+    open(snapshot_vmstate, "w+").close()
+
+    # Hardlink the snapshot files into the microvm jail.
+    jailed_mem = vm.create_jailed_resource(snapshot_mem)
+    jailed_vmstate = vm.create_jailed_resource(snapshot_vmstate)
+
+    vm.create_tap_and_ssh_config(host_ip=DEFAULT_HOST_IP,
+                                 guest_ip=DEFAULT_GUEST_IP,
+                                 netmask_len=DEFAULT_NETMASK,
+                                 tapname=DEFAULT_TAP_NAME)
+
+    # Get the Firecracker process starting time
+    fc_time, err = _get_process_start_time(vm.jailer_clone_pid)
+    assert err == 0
+    logger.info("FC process time before snapshoat load: %s.", fc_time)
+
+    # Load the snapshot
+    response = vm.snapshot_load.put(mem_file_path=jailed_mem,
+                                    snapshot_path=jailed_vmstate)
+
+    logger.info("Response status code %d, content: %s.",
+                response.status_code,
+                response.text)
+    assert vm.api_session.is_status_bad_request(response.status_code)
+
+    # Check if FC process is closed
+    seconds = 0.1
+    fc_exit_time, err = _get_process_start_time(vm.jailer_clone_pid)
+    while seconds < 300:
+        if err != 0 or fc_time != fc_exit_time:
+            break
+        seconds += 0.1
+        time.sleep(0.1)
+        fc_exit_time, err = _get_process_start_time(vm.jailer_clone_pid)
+    assert err != 0 or fc_time != fc_exit_time, "Firecracker process should \
```

Comment:
nit:
leave a helpful comment for future generations
```suggestion
    # Check that either:
    #  - pid no longer exists, or
    #  - pid exists but is a different process (different process start time)
    assert err != 0 or pid_time != fc_time, "Firecracker process should \
```

Reply:
Done